[Yang Jiyu] iP

YangJiyu98 opened the pull request.

Commit:
Add Gradle support

YangJiyu98 committed:
Level-1: Greet, Echo, Exit

YangJiyu98 committed:
Level 2. Add, List

YangJiyu98 committed:
Level-3: Mark as Done

YangJiyu98 committed:
Level-4: ToDo, Event, Deadline. Done w inheritance

YangJiyu98 committed:
A-TextUiTesting: Automated Text UI Testing

YangJiyu98 committed:
Level-5: Handle Errors

YangJiyu98 committed:
Level-6: Delete

YangJiyu98 committed:
Level 7: Save

YangJiyu98 committed:
Update regex in database to split by // instead of /

YangJiyu98 committed:
Level-8: Dates and Times

YangJiyu98 committed:
Merge branch 'branch-level-7' into master

Merge branch-level-7 without fast forwarding

YangJiyu98 committed:
Merge branch 'branch-level-8' into master

Merge branch-level-8 with no fast forwarding

YangJiyu98 committed:
Update methods that write to data base

YangJiyu98 committed:
A-MoreOOP: Use More OOP

Add class Parser, Storage, TasKList Ui.

Things to follow
- Parser not able to handle general commands yet.
- Ui still can handle more specific messages
- can use Duke object instead of writing in Main class

YangJiyu98 committed:
A-Packages: Organize into Packages

Organize classes into dukeclass package

YangJiyu98 committed:
A-Junit: Add JUnit Tests

Add JUnit test for Storage and Parser

YangJiyu98 committed:
Fix Ui bug

YangJiyu98 committed:
A-JavaDoc: JavaDoc

Add JavaDoc comments

YangJiyu98 committed:
A-CodingStandard: Follow the Coding Standard

Change some import statments to follow coding standard.

YangJiyu98 committed:
Level-9: Find

Add Find capabilities. Limited to single words only.

YangJiyu98 committed:
Merge branch 'branch-A-JavaDoc' into master

YangJiyu98 committed:
Merge branch 'branch-A-CodingStandard' into master

YangJiyu98 committed:
Merge branch 'branch-Level-9' into master

YangJiyu98 committed:
Add more OOP to Parser and Storage class

YangJiyu98 committed:
Add more OOP to Parser and change naming

Change naming to meetnaming conventions

YangJiyu98 committed:
Update code to coding standards

YangJiyu98 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

YangJiyu98 committed:
Add gradle to automate project builds
Add checkstyle
Update code formatting to meet checkstyle standards

YangJiyu98 committed:
Merge branch 'A-Gradle' into master

T-Fang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,61 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import dukeclass.Parser;
+import dukeclass.Storage;
+import dukeclass.TaskList;
+import dukeclass.Ui;
```
When we want to import more than three components from the same package, is it better to use "import dukeclass.*"?

ianyong replied:
Hi, I don't think using wildcard imports is better. The following line can be found in the CS2103T Java coding standard:
> Imported classes should always be listed explicitly.

T-Fang replied:
Sorry, I think I have remembered it wrongly :( Thanks!

T-Fang reviewed (commented):
Great work! You have followed most of the guidelines!

YangJiyu98 committed:
Add JavaFx

JavaFx not able to fully implement Duke as in CLI. Work in progress.

ianyong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,61 @@
-        System.out.println("Hello from\n" + logo);
+
+    /**
+     * Duke to be started, create files for storage, run commands and write to files.
```
Perhaps this could be rewritten to be in line with the coding standard?

> In method header comments, the first sentence should start in the form `Returns ...`, `Sends ...`, `Adds ...` (not `Return` or `Returnning` etc.)

ianyong commented on src/main/java/dukeclass/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Override
+    public String toString() {
```
Consider adding header comments for all public methods as suggested in the style guide; there are a few public methods which do not have comments.

ianyong commented on test/dukeclass/StorageTest.java:
```diff
@@ -0,0 +1,31 @@
+package dukeclass;
+
+import dukeclass.*;
```
Should the imports be listed explicitly instead?

ianyong commented on src/main/java/dukeclass/Storage.java:
```diff
@@ -0,0 +1,138 @@
+     * @return File for storage of TaskList.
+     */
+    public static File createFile() {
+        File d = new File("data");
```
Perhaps a more descriptive variable name could be used since the scope of this variable is not that small? Same for some of the other one letter variables.

ianyong commented on src/main/java/dukeclass/Ui.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @return welcome message of String type.
+     */
+    public static String welcomeMessage() {
```
Would it be possible to rewrite this as a verb?

ianyong reviewed (commented):
Looks good other than a few nits!

YangJiyu98 committed:
Update GUI to include Duke, cannot run on CLI

YangJiyu98 committed:
Update Duke GUI

Add "help" command
Update main GUI to be responsive when scaling
Update textfield if blank cannot Send
Update dialouge box GUI to expand when text exceeds preferred size
Bug on VBox in scroll pane, not responsive
Update ticks and crosses for task status

YangJiyu98 committed:
Add extenstion for snooze

YangJiyu98 committed:
Merge pull request #1 from YangJiyu98/add-extension

Add extenstion for snooze

YangJiyu98 committed:
Add assertion statements

YangJiyu98 committed:
Merge pull request #2 from YangJiyu98/branch-A-Assertions

Add assertion statements

YangJiyu98 committed:
Update code quality

YangJiyu98 committed:
Merge pull request #3 from YangJiyu98/branch-A-CodeQuality

Update code quality

YangJiyu98 committed:
Add User guide

Add user guide
Update database parser
Update string output for event and deadline task
Update GUI

YangJiyu98 committed:
Replace Ui screenshot

YangJiyu98 committed:
Update test paths on StorageTest
